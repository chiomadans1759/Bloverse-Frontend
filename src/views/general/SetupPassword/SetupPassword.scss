@import 'sass-rem';

#bl-setup-password {
  height: 100vh;
  .main-content {
    width: rem(281px);
    padding: 0;
    padding-top: rem(70px);
    .setup-password-form {
      margin: rem(10px) 0;
      justify-content: flex-start;
      &>form {
        width: 100%;
      }
      button.bl-form-btn {
        background-image: linear-gradient( to right, #003CE1, #2F80ED);
      }
      input, button.bl-form-btn {
        width: 100%;
        border-radius: rem(2px) !important;
        font-size: rem(14px);
      }
      input::placeholder {
        color: #B7B7B7;
        font-size: rem(14px);
      }
    }
    .title {
      white-space: nowrap; 
      overflow: hidden;
      text-overflow: ellipsis;
      &>h3 {
        color: #252525;
        text-transform: capitalize;
        text-align: justify;
        font-size: rem(24px);
      }
      &>p {
        text-align: justify;
        line-height: 2.4;
        font-size: rem(14px);
      }
    }
  }
  .footer-container {
    position: fixed;
    bottom: 0;
    color: #B7B7B7;
    line-height: rem(40px);
  }
}
