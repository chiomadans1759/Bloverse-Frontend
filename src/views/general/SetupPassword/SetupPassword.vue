<template>
  <main id="bl-setup-password" class="container-fluid p-0 bl-setup-password">
    <navbar :showBackArrow="showBackArrow" :showNavigations="showNavigations" customClass="back-nav"/>
    <div class="container main-content">
      <div class="title">
        <h3>Password</h3>
        <p>Enter your password</p>
      </div>
      <div class="row setup-password-form">
        <form>
          <div class="form-group">
            <input type="password" class="form-control" id="password"
              aria-describedby="Password" placeholder="Password"
            >
          </div>
          <div class="form-group">
            <input type="password" class="form-control"
              id="confirm-password" placeholder="Confirm Password"
            >
          </div>
          <button type="button" class="btn btn-primary btn-block bl-form-btn"
            data-toggle="modal" href="#profile-details-modal"
          >
            Save
          </button>
        </form>
      </div>
    </div>
    <div class="footer-container">
      <p><span class="bloverse">bloverse</span> 2019. All Rights Reserved</p>
    </div>

    <modal
      target="profile-details-modal"
      altTitle="Profile Details"
      dark
      noBorderRadius
    >
      <profile-details />

      <profile-details-footer slot="footer" />
    </modal>

  </main>
</template>

<script>
import Navbar from '@/components/Navbar/Navbar.vue';
import Modal from '@/components/Modal/Modal.vue';
import ProfileDetails from '@/components/ProfileDetails/ProfileDetails.vue';
import ProfileDetailsFooter from '@/components/ProfileDetails/ProfileDetailsFooter.vue';

export default {
  name: 'SetupPassword',
  components: { Navbar, Modal, ProfileDetails, ProfileDetailsFooter },
  data() {
    return {
      showNavigations: false,
      showBackArrow: true
    }
  }
}
</script>

<style lang="scss" scoped>
 @import './SetupPassword.scss';
</style>
