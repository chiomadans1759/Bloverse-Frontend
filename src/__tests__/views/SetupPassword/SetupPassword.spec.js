import { mount, RouterLinkStub } from '@vue/test-utils';
import SetupPassword from '@/views/general/SetupPassword/SetupPassword.vue';

describe('SetupPassword Component', () => {
  it('should render correctly', () => {

    const wrapper = mount(SetupPassword, {
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
