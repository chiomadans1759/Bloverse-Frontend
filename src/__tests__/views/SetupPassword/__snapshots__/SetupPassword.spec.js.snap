// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetupPassword Component should render correctly 1`] = `
<main
  class="container-fluid p-0 bl-setup-password"
  id="bl-setup-password"
>
  <header
    id="feeds-header"
  >
    <nav
      class="navbar navbar-expand-lg navbar-light px-5"
    >
      <div
        class="container-fluid back-nav"
      >
        <a
          class="backlink mr-4"
        >
          <div
            class="btn rounded-btn-circle d-flex justify-content-center"
          >
            <i
              class="fal fa-arrow-left back-icon"
            />
          </div>
        </a>
         
        <span
          class="back-text"
        >
          Back
        </span>
         
        <a
          class="router-link"
        >
          <img
            class="nav-logo"
            src="@/assets/Bloverse.svg"
          />
           
          <!---->
        </a>
         
        <!---->
         
        <!---->
      </div>
    </nav>
     
    <main
      id="bl-modal"
    >
      <div
        aria-hidden="true"
        aria-labelledby="customModal"
        class="modal fade"
        id="feedback-modal"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="modal-dialog modal-dialog-centered"
          role="document"
        >
          <div
            class="modal-content"
          >
            <div
              class="modal-header darkBackground"
            >
              <!---->
               
              <div>
                <h6>
                  <strong
                    class="alt-title"
                  >
                    Send Us Some Feedback
                  </strong>
                </h6>
                 
                <h6
                  class="mt-2 sub-title"
                >
                  Found a bug? Have a suggestion?. Fill out the form below and we'll take a look!
                </h6>
              </div>
               
              <button
                aria-label="Close"
                class="close"
                data-dismiss="modal"
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  ×
                </span>
              </button>
            </div>
             
            <div
              class="modal-body"
            >
              <div
                class="bl-modal-content"
              >
                <p>
                  Enter your feedback here!
                </p>
                 
                <textarea
                  class="form-control mb-2 p-1"
                />
                 
                <label
                  class="ml-1 mr-3"
                >
                  <input
                    name="feedback-type"
                    type="radio"
                  />
                   Bug
      
                </label>
                 
                <label
                  class="mr-3"
                >
                  <input
                    name="feedback-type"
                    type="radio"
                  />
                   Comment
      
                </label>
                 
                <label
                  class="mr-3"
                >
                  <input
                    name="feedback-type"
                    type="radio"
                  />
                   Other
      
                </label>
              </div>
            </div>
             
            <div
              class="modal-footer"
            >
              <button
                class="btn btn-outline-primary"
                data-dismiss="modal"
                type="button"
              >
                
              Close
            
              </button>
               
              <button
                class="btn btn-primary"
                type="button"
              >
                Send Feedback
              </button>
            </div>
          </div>
        </div>
      </div>
    </main>
  </header>
   
  <div
    class="container main-content"
  >
    <div
      class="title"
    >
      <h3>
        Create Password
      </h3>
       
      <p>
        Enter your password
      </p>
    </div>
     
    <div
      class="row setup-password-form"
    >
      <form>
        <div
          class="form-group"
        >
          <input
            aria-describedby="Password"
            class="form-control"
            id="password"
            placeholder="Password"
            type="password"
          />
        </div>
         
        <div
          class="form-group"
        >
          <input
            class="form-control"
            id="confirm-password"
            placeholder="Confirm Password"
            type="password"
          />
        </div>
         
        <button
          class="btn btn-primary btn-block bl-form-btn"
          data-toggle="modal"
          href="#profile-details-modal"
          type="button"
        >
          
          Save
        
        </button>
      </form>
    </div>
  </div>
   
  <div
    class="footer-container"
  >
    <p>
      <span
        class="bloverse"
      >
        bloverse
      </span>
       2019. All Rights Reserved
    </p>
  </div>
   
  <main
    id="bl-modal"
  >
    <div
      aria-hidden="true"
      aria-labelledby="customModal"
      class="modal fade"
      id="profile-details-modal"
      role="dialog"
      tabindex="-1"
    >
      <div
        class="modal-dialog modal-dialog-centered"
        role="document"
      >
        <div
          class="modal-content noBorderRadius"
        >
          <div
            class="modal-header darkBackground"
          >
            <!---->
             
            <div>
              <h6>
                <strong
                  class="alt-title"
                >
                  Profile Details
                </strong>
              </h6>
               
              <!---->
            </div>
             
            <button
              aria-label="Close"
              class="close"
              data-dismiss="modal"
              type="button"
            >
              <span
                aria-hidden="true"
              >
                ×
              </span>
            </button>
          </div>
           
          <div
            class="modal-body"
          >
            <div
              class="profile-details-content"
            >
              <div
                class="contact-details mb-4"
              >
                <p>
                  John Doe
                </p>
                 
                <p>
                  <i
                    class="fal fa-phone mr-2 contact-icon"
                  />
                  
      +234 000 000 0000
    
                </p>
                 
                <p>
                  <i
                    class="fal fa-envelope mr-2 contact-icon"
                  />
                  
      email@bloverse.com
    
                </p>
                 
                <p>
                  <i
                    class="fal fa-map-marker-alt mr-2 contact-icon"
                  />
                  
      Lagos, Nigeria
    
                </p>
              </div>
               
              <main
                class="category-card"
              >
                <div
                  class="header d-flex justify-content-between "
                >
                  <h5
                    class="profile-details-title"
                  >
                    <strong>
                      Category
                    </strong>
                  </h5>
                   
                  <!---->
                </div>
                 
                <div
                  class="mt-1"
                >
                  <!---->
                   
                  <div
                    class="category-list d-flex flex-wrap mt-2"
                  >
                    <div
                      class="category border rounded my-1 mr-2 p-2"
                    >
                      <span>
                        Category One
                      </span>
                    </div>
                     
                    <div
                      class="category border rounded my-1 mr-2 p-2"
                    >
                      <span>
                        Category Two
                      </span>
                    </div>
                     
                    <div
                      class="category border rounded my-1 mr-2 p-2"
                    >
                      <span>
                        Category Three
                      </span>
                    </div>
                     
                    <div
                      class="category border rounded my-1 mr-2 p-2"
                    >
                      <span>
                        Category Four
                      </span>
                    </div>
                     
                    <div
                      class="category border rounded my-1 mr-2 p-2"
                    >
                      <span>
                        Category Five
                      </span>
                    </div>
                     
                    <div
                      class="category border rounded my-1 mr-2 p-2"
                    >
                      <span>
                        Category Six
                      </span>
                    </div>
                     
                    <div
                      class="category border rounded my-1 mr-2 p-2"
                    >
                      <span>
                        Category Seven
                      </span>
                    </div>
                  </div>
                </div>
              </main>
               
              <div
                class="my-4"
                id="bl-socail-media"
              >
                <p
                  class="mb-2 social-title"
                >
                  <strong>
                    Social Media
                  </strong>
                </p>
                 
                <div
                  class="d-flex flex-wrap"
                >
                  <div
                    class="mr-2"
                  >
                    <div
                      class="border rounded p-2 mb-1 d-flex align-items-center"
                    >
                      <img
                        alt="facebook"
                        class="social-icon"
                        src="@/assets/facebook-square.svg"
                      />
                       
                      <h6
                        class="social-text text-center pl-2"
                      >
                        Facebook
                      </h6>
                    </div>
                  </div>
                   
                  <div
                    class="mr-2"
                  >
                    <div
                      class="border rounded p-2 mb-1 d-flex align-items-center"
                    >
                      <img
                        alt="facebook"
                        class="social-icon"
                        src="@/assets/twitter-square.svg"
                      />
                       
                      <h6
                        class="social-text text-center pl-2"
                      >
                        Twitter
                      </h6>
                    </div>
                  </div>
                   
                  <div
                    class="mr-2"
                  >
                    <div
                      class="border rounded p-2 mb-1 d-flex align-items-center"
                    >
                      <img
                        alt="facebook"
                        class="social-icon"
                        src="@/assets/instagram-square.svg"
                      />
                       
                      <h6
                        class="social-text text-center pl-2"
                      >
                        Instagram
                      </h6>
                    </div>
                  </div>
                   
                  <div
                    class="mr-2"
                  >
                    <div
                      class="border rounded p-2 mb-1 d-flex align-items-center"
                    >
                      <img
                        alt="facebook"
                        class="social-icon"
                        src="@/assets/linkedin-square.svg"
                      />
                       
                      <h6
                        class="social-text text-center pl-2"
                      >
                        Linkedin
                      </h6>
                    </div>
                  </div>
                </div>
              </div>
            </div>
             
          </div>
           
          <div
            class="modal-footer"
          >
            <main
              class="row mx-0 profile-details-footer"
            >
              <div
                class="col-5 col-md-8 px-0 skip"
              >
                <a
                  data-dismiss="modal"
                >
                  Skip
                </a>
              </div>
               
              <a
                class="btn btn-primary btn-block col-7 col-md-4 complete-profile-btn"
                data-dismiss="modal"
                type="link"
              >
                
    Complete profile
  
              </a>
            </main>
          </div>
        </div>
      </div>
    </div>
  </main>
</main>
`;
